xbmc/messaging/ApplicationMessenger.cpp

```diff
@@ -225,15 +225,13 @@ void CApplicationMessenger::ProcessMessage(ThreadMessage *pMsg)
     return;
   }
 
-  CSingleLock lock(m_critSection);
   int mask = pMsg->dwMessage & TMSG_MASK_MESSAGE;
 
+  CSingleLock lock(m_targetCritSection);
```

Comment:
```
  IMessageTarget* target = nullptr;
  {
    CSingleLock lock(m_targetCritSection);
    auto targetI = m_mapTargets.find(mask);
    if (targetI != m_mapTargets.end())
    {
      target = *targetI;
    }
  }
```

`map.at()` throws an exception if the key is not in the map

Reply:
Yes

Reply:
Does not work. I think it needs to be  `target = targetI->second;`